app/src/main/java/ch/hikemate/app/model/route/HikeRoutesRepositoryOverpass.kt

```diff
@@ -104,14 +107,102 @@ class HikeRoutesRepositoryOverpass(val client: OkHttpClient) : HikeRoutesReposit
+     */
+    private fun flattenHikeWays(hikeWays: List<HikeWay>): HikeWay {
+      val points = mutableListOf<LatLong>()
+      for (i in hikeWays.indices) {
+        if (points.isEmpty()) {
+          handleFirstWay(hikeWays, points, i)
+        } else {
+          handleSubsequentWays(hikeWays, points, i)
+        }
+      }
+      return points
+    }
+
+    /**
+     * Procedure to handles the first way of the hike during the flattening process.
+     *
+     * @param hikeWays The list of hike ways.
+     * @param points The list of points to add the way to.
+     * @param i The index of the way to handle.
+     */
+    private fun handleFirstWay(hikeWays: List<HikeWay>, points: MutableList<LatLong>, i: Int) {
+      if (i < hikeWays.size - 1 &&
+          (hikeWays[i].first() == hikeWays[i + 1].first() ||
+              hikeWays[i].first() == hikeWays[i + 1].last())) {
+        points.addAll(hikeWays[i].reversed())
+      } else {
+        points.addAll(hikeWays[i])
+      }
+    }
+
+    /**
+     * Procedure to handles the subsequent ways of the hike during the flattening process.
+     *
+     * @param hikeWays The list of hike ways.
+     * @param points The list of points to add the way to.
+     * @param i The index of the way to handle.
+     */
+    private fun handleSubsequentWays(
+        hikeWays: List<HikeWay>,
+        points: MutableList<LatLong>,
+        i: Int
+    ) {
+      when {
+        hikeWays[i].first() == points.last() ->
+            points.addAll(hikeWays[i].subList(1, hikeWays[i].size))
+        hikeWays[i].last() == points.last() ->
+            points.addAll(hikeWays[i].reversed().subList(1, hikeWays[i].size))
+        else -> handleUnconnectedWays(hikeWays, points, i)
+      }
+    }
+
+    /**
+     * Procedure to handles the unconnected ways of the hike during the flattening process.
+     *
+     * @param hikeWays The list of hike ways.
+     * @param points The list of points to add the way to.
+     * @param i The index of the way to handle.
+     */
+    private fun handleUnconnectedWays(
+        hikeWays: List<HikeWay>,
+        points: MutableList<LatLong>,
+        i: Int
+    ) {
+      val lastPoint = points.last()
+      val firstWayPoint = hikeWays[i].first()
+      val lastWayPoint = hikeWays[i].last()
+      val distanceFirst = firstWayPoint.distanceTo(lastPoint)
+      val distanceLast = lastWayPoint.distanceTo(lastPoint)
+
+      when {
+        distanceFirst < distanceLast && distanceFirst < MAX_DISTANCE_BETWEEN_OPENED_PATH ->
+            points.addAll(hikeWays[i])
+        distanceFirst >= distanceLast && distanceLast < MAX_DISTANCE_BETWEEN_OPENED_PATH ->
+            points.addAll(hikeWays[i].reversed())
+        else -> points.clear() // Discard the hikes if unconnected
+      }
+    }
+
     /**
      * Parses the routes from the Overpass API response.
      *
      * @param responseReader The reader for the response body.
      * @return The list of parsed routes.
```

Comment:
Nitpick: Perhaps indicate in the description that the list may be null.

Reply:
I don't understand what you meant. The list cannot be null. It can be empty but never null.